test/emu_mps/test_entanglement_entropy.py

```diff
@@ -0,0 +1,117 @@
+import pulser
+import emu_mps
+from emu_mps.observables import EntanglementEntropy
+
+
+def create_constant_pulse_sequence():
+    # Choosing the device
+    device = pulser.AnalogDevice
+
+    # creating the register
+    square_length = 3
+    omega = 12
+    delta = 4
+    R_interatomic = pulser.AnalogDevice.rydberg_blockade_radius(2 * omega)
+    register = pulser.Register.square(square_length, R_interatomic, prefix="q")
+
+    # creating the sequence
+    seq = pulser.Sequence(register, device)
+    seq.declare_channel("rydberg_global", "rydberg_global")
+
+    # adding constant pulses to the sequence
+    pulse1 = pulser.Pulse.ConstantPulse(100, 0, 0, 0)
+    pulse2 = pulser.Pulse.ConstantPulse(200, 0.5 * omega, 0.2 * delta, 0)
+    pulse3 = pulser.Pulse.ConstantPulse(200, 0.2 * omega, 0.3 * delta, 0)
+    seq.add(pulse1, "rydberg_global")
+    seq.add(pulse2, "rydberg_global")
+    seq.add(pulse3, "rydberg_global")
+
+    return seq
+
+
+def test_entanglement_entropy_generic_case():
+    """
+    the S_E should be returned as a float
+    testing that the new EntanglementEntropy subclass is working
+    """
+    initial_state = emu_mps.MPS.from_state_amplitudes(
+        eigenstates=("r", "g"), amplitudes={"rrrrrrrrr": 1.0}
+    )
+    evaluation_times = [1 / 100, 10 / 100, 20 / 100, 50 / 100, 1.0]
+    ent_entropy = EntanglementEntropy(bond_index=5, evaluation_times=evaluation_times)
+    config = emu_mps.MPSConfig(observables=[ent_entropy], initial_state=initial_state)
+    seq = create_constant_pulse_sequence()
+
+    backend = emu_mps.MPSBackend(sequence=seq, config=config)
+    result = backend.run()
+
+    result_times = result.get_result_times(observable=ent_entropy)
+    assert 0.5 in result_times
+
+    entropy = result.get_result(observable=ent_entropy, time=1)
+    assert isinstance(entropy, float)
+
+
+def test_zero_entropy_product_state():
+    """
+    starting from a product state, the S_E should be zero across all bonds
+    """
+    initial_state = emu_mps.MPS.from_state_amplitudes(
+        eigenstates=("r", "g"), amplitudes={"rrrrrrrrr": 1.0}
+    )
+    entropies = []
+    for b in range(1, 9):
+        ent_entropy = EntanglementEntropy(bond_index=b, evaluation_times=[1 / 100])
+        config = emu_mps.MPSConfig(observables=[ent_entropy], initial_state=initial_state)
+        seq = create_constant_pulse_sequence()
+        backend = emu_mps.MPSBackend(sequence=seq, config=config)
+        result = backend.run()
+        S_E_res = result.get_result(ent_entropy, time=1 / 100)
+        entropies.append(S_E_res)
+
+    assert all(s < 1e-7 for s in entropies)
```

Comment:
this somehow implies that entropy is a float, because you comparing it with floats. I would just remove the test above